operator/internal/ymlutils/extractor.go

```diff
@@ -0,0 +1,115 @@
+package ymlutils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	expectedLenAfterSplit = 2
+)
+
+func GatherChartGvks(chartPath string) ([]schema.GroupVersionKind, error) {
+	var allGvks []schema.GroupVersionKind
+	appendToSlice := func(gvk schema.GroupVersionKind) {
+		if reflect.DeepEqual(gvk, schema.GroupVersionKind{}) {
+			return
+		}
+		for _, v := range allGvks {
+			if reflect.DeepEqual(gvk, v) {
+				return
+			}
+		}
+		allGvks = append(allGvks, gvk)
+	}
+
+	root := fmt.Sprintf("%s/templates/", chartPath)
+	if err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		isNotYamlFileExtension := !strings.HasSuffix(info.Name(), ".yml") && !strings.HasSuffix(info.Name(), ".yaml")
+		if isNotYamlFileExtension {
+			return nil
+		}
+
+		bytes, err := os.ReadFile(fmt.Sprintf("%s/%s", root, info.Name()))
+		if err != nil {
+			return err
+		}
+
+		fileGvks, err := ExtractGvkFromYml(string(bytes))
+		if err != nil {
+			return err
+		}
+
+		for _, gvk := range fileGvks {
+			appendToSlice(gvk)
+		}
+
+		return nil
+	}); err != nil {
+		return []schema.GroupVersionKind{}, err
+	}
+
+	return allGvks, nil
+}
+
+func ExtractGvkFromYml(wholeFile string) ([]schema.GroupVersionKind, error) {
+	var gvks []schema.GroupVersionKind
+	parts := strings.Split(wholeFile, "---\n")
+	for _, part := range parts {
+		if part == "" {
+			continue
+		}
+		var yamlGvk schema.GroupVersionKind
+		lines := strings.Split(part, "\n")
+		for _, line := range lines {
+			if strings.HasPrefix(line, "apiVersion:") {
+				yamlGvk.Version = strings.TrimSpace(strings.Split(line, ":")[1])
+			}
+			if strings.HasPrefix(line, "kind:") {
+				yamlGvk.Kind = strings.TrimSpace(strings.Split(line, ":")[1])
+			}
+		}
+		if yamlGvk.Kind != "" && yamlGvk.Version != "" {
```

Comment:
we could check apiVersion, kind string here as suggested in comment for L71 and build `GroupVersionKind` struct when required using `schema.ParseGroupVersion` and `gv.WithKind`